packages/components/src/toolbar/toolbar-item/index.tsx

```diff
@@ -15,8 +14,18 @@ import warning from '@wordpress/warning';
  * Internal dependencies
  */
 import ToolbarContext from '../toolbar-context';
+import type { ToolbarItemProps } from './types';
+import type { WordPressComponentProps } from '../../ui/context';
 
-function ToolbarItem( { children, as: Component, ...props }, ref ) {
+function ToolbarItem(
+	{
+		children,
+		as,
+		...props
+	}: WordPressComponentProps< ToolbarItemProps, 'div', false >,
+	ref: ForwardedRef< any >
+) {
+	const Component = as;
```

Comment:
No reason. My logic was doing destructuring with a capitalized name, ex for a class or JSX.Element, always leads me to think its a type declaration, so I try to avoid it. Happy to leave it in the function declaration I personally think it can be confsing.

Reply:
Interesting, never thought of that angle! I guess one downside of moving it to a separate `const` is that the `as` variable will continue to be in scope, potentially causing inconsistencies where some lines of code refer to `Component` and others to `as`.